### PR TITLE
feat: RIP-7711 basic implementation

### PR DESCRIPTION
**Pull request by kangsorang**
# Description

RIP-7711 basic implementation.
(https://github.com/ethereum/RIPs/blob/master/RIPS/rip-7711.md)

### CODE REVIEW

File: core/types/transaction.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ const (
 	DynamicFeeTxType = 0x02
 	BlobTxType       = 0x03
 	Rip7560Type      = 0x04
+	// TODO: do we really need a new type?
+	Rip7560BundleHeaderType = 0x05
```
Comment (sm-stack) [inline core/types/transaction.go:55]: I think we'd better discuss this w/ 4337 team about the intention of the spec.

File: core/types/transaction_marshalling.go [hunk 2]
```diff
@@ -591,6 +602,29 @@ func (tx *Transaction) UnmarshalJSON(input []byte) error {
 			return errors.New("missing required field 'value' in transaction")
 		}
 		itx.Value = (*big.Int)(dec.Value)
+	case Rip7560BundleHeaderType:
+		log.Info("Unmarshal Rip756BundleHeaderType JSON")
+		log.Info(string(debug.Stack()))
```
Comment (sm-stack) [inline core/types/transaction_marshalling.go]: Should we remain this logging here?

File: internal/ethapi/api.go [hunk 3]
```diff
@@ -1583,6 +1584,10 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 		result.PaymasterGas = (hexutil.Uint64)(tx.PaymasterGas())
 		result.PostOpGas = (hexutil.Uint64)(tx.PostOpGas())
 		result.BigNonce = (*hexutil.Big)(tx.BigNonce())
+	case types.Rip7560BundleHeaderType:
+		log.Info("rip7560(bundleHeader) new RpcTransaction")
+		result.ChainID = (*hexutil.Big)(tx.ChainId())
+		result.TransactionCount = (hexutil.Uint64)(tx.TransactionCount())
```
Comment (sm-stack) [inline internal/ethapi/api.go]: Do we need changes here? This is not a tx crafted or received by RPC, is it?

File: miner/worker.go [hunk 4]
```diff
@@ -1111,6 +1111,15 @@ func (w *worker) commitRip7560TransactionsBundle(env *environment, txs *types.Ex
...
 	}
 
+	// adding RIP-7560 bundle header transaction here
+	bundleHeaderTx := types.NewTx(&types.Rip7560BundleHeaderTx{
+		ChainID: w.chainConfig.ChainID,
+		// TODO: if some transactions are invalid, do we need to drop the bundle? or fix TransactionCount at that time?
+		TransactionCount: 1, //uint64(len(txs.Transactions)),
+		BlockNumber:      env.header.Number,
+		TransactionIndex: 0,
+	})
```
Comment (sm-stack) [inline miner/worker.go]: Maybe crafting this `bundleHeaderTx` after validating and executing all txs, and then append it ahead of `validatedTxs` can work, how do you think about this?

File: core/types/transaction_marshalling.go (hunk 2 above)
Comment (kangsorang) [reply core/types/transaction_marshalling.go]: Ok, I'll exclude the related logs.

File: internal/ethapi/api.go (hunk 3 above)
Comment (kangsorang) [reply internal/ethapi/api.go]: Required to configure responses to RPC requests. 
For example, this part is needed when serializing to an RPC representation format for the response of a request like getBlockByHash.

Comment (sm-stack) [reply internal/ethapi/api.go]: Oh I missed that part, thanks.

File: miner/worker.go (hunk 4 above)
Comment (kangsorang) [reply miner/worker.go]: Yes, that's a good idea. I've modified the structure to reflect your feedback.

File: core/types/transaction.go (hunk 1 above)
Comment (kangsorang) [reply core/types/transaction.go:55]: Yeah, that's a good idea

**Review by sm-stack — APPROVED** [review]
LGTM

**Pull request merged**